src/ImageSharp/Common/Extensions/StreamExtensions.cs

```diff
@@ -82,5 +83,22 @@ public static void Write(this Stream stream, IManagedByteBuffer buffer)
         {
             stream.Write(buffer.Array, 0, buffer.Length());
         }
+
+#if NET472 || NETSTANDARD1_3 || NETSTANDARD2_0
+        // This is a port of the CoreFX implementation and is MIT Licensed: https://github.com/dotnet/coreclr/blob/c4dca1072d15bdda64c754ad1ea474b1580fa554/src/System.Private.CoreLib/shared/System/IO/Stream.cs#L770
+        public static void Write(this Stream stream, ReadOnlySpan<byte> buffer)
+        {
+            byte[] sharedBuffer = ArrayPool<byte>.Shared.Rent(buffer.Length);
```

Comment:
We usually prefer using our own memory management primitives like [`MemoryAllocator.AllocateManagedByteBuffer()`](https://github.com/SixLabors/Core/blob/master/src/SixLabors.Core/Memory/MemoryAllocator.cs#L29) instead of `ArrayPool.Shared`.

Reply:
This is matching a signature and implementation available in .NET Core. `ArrayPool<byte>.Shared` should generally be allocation-less for something like this (not necessarily for other `T`) since most of the framework is fairly dependent on it.

I'll run benchmarks on net472 as well though, in order to see what the overhead here is, if any, since it isn't applicable to netcoreapp2.1 (where the above benchmark was run).

Reply:
Our "allocator" is typically also an allocation free pool, but we definitely have an overhead because of nested virtual calls, and other infrastructure stuff.
Everything depends on the size of the array. It's not uncommon to have very large (> 1 MB) buffers in image processing, which works better with our allocator/pool in my experience, but for those buffers we should avoid invoking this extension method anyways.

Reply:
Definitely and I don't recall if benchmark.net isolates each iteration or not (I know it does for separate tests), so it could be that the benchmark is "lying" for full framework and it is hiding the allocation cost (so it might not be representative of real world scenarios).

It would be possible to take a `MemoryAllocator` as an optional parameter or to not have `SosHeaderYCbCr` take advantage of this compiler optimization (the single copy on class instantiation isn't terrible, but losing out on permanent pinning is a bit unfortunate) or to only not use the optimization on full framework if there are concerns about this regressing full framework.

IIRC, the `ArrayPool<byte>.Shared` is optimized for up to 2MB, since we use it for things like the JSON and XML readers.

Reply:
Let's add a comment explaining why we use that pool (signature matching) and move on.

Reply:
Fixed. Also rebased onto the current HEAD.